source/Ocl/Parsing/HeredocParser.cs

```diff
@@ -63,11 +63,11 @@ class HeredocParser
 
         internal static Parser<string> End(string tag)
         {
-            var endLine = from leadingWhitespace in ParserCommon.WhitespaceExceptNewline.Many()
+            var endLine = from leadingWhitespace in ParserCommon.WhitespaceExceptNewline.Many().Text()
                 from endtag in Parse.String(tag).Text()
-                from trailingWhitespace in ParserCommon.WhitespaceExceptNewline.Many()
-                select endtag;
-
+                from trailingWhitespace in ParserCommon.WhitespaceExceptNewline.Many().Optional()
+                select leadingWhitespace + endtag;
```

Comment:
By including the leading whitespace we [can later factor it into the calculation](https://github.com/OctopusDeploy/Ocl/pull/131/files#diff-351f72ce42f3761c42324f1d514c550d5dfcbe419d9189dc28e4c8162fc0c78fR105) for the de-indent calculation